lib/kang.js

```diff
@@ -80,20 +85,42 @@ function knMakeSource(str)
 		str = str.substr(0, c);
 	}
 
-	c = str.indexOf(':');
-	if (c == -1) {
-		host = str;
+	if ((matched = IPV4_HOST.exec(str)) !== null) {
```

Comment:
[RFC 2732](https://www.ietf.org/rfc/rfc2732.txt) updates the grammar for URLs to describe how to handle literal IPv6 addresses. I can update the PR to include updated documentation on the URL format.

A common library would probably be good. I don't know if there's another URL parsing library out there that would work well here, or if we should have our own.